src/shared/approval-status/use-server-date-time-as-local.test.js

```diff
@@ -0,0 +1,126 @@
+import { useConfig } from '@dhis2/app-runtime'
+import { useServerDateTimeAsLocal } from './use-server-date-time-as-local.js'
+
+jest.mock('@dhis2/app-runtime', () => ({
+    useConfig: jest.fn(),
+}))
+
+describe('useServerDateTimeAsLocal hook', () => {
+    /**
+     * Client timezone is set to Etc/UTC when starting the test runner
+     * See `package.json`:
+     * "test": "TZ=Etc/UTC d2-app-scripts test",
+     */
+    it('Converts correctly if on server time', () => {
+        useConfig.mockReturnValueOnce({
+            systemInfo: { serverTimeZoneId: 'Etc/UTC' },
+        })
+        const dateStr = '2021-08-31 16:00:00'
+        const localDateTime = new Date(dateStr)
+        const converted = useServerDateTimeAsLocal(dateStr)
+        expect(converted.getHours()).toEqual(localDateTime.getHours())
+    })
+
+    describe('timezone IDs unadjusted for daylight saving times', () => {
+        it('Converts correctly if ahead of server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Etc/GMT-9' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            expect(converted.getHours()).toEqual(localDateTime.getHours() - 9)
+        })
+
+        it('Converts correctly if behind on server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Etc/GMT+4' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            expect(converted.getHours()).toEqual(localDateTime.getHours() + 4)
+        })
+    })
+
+    describe('timezone IDs adjusted for daylight saving times', () => {
+        it('Converts correctly if ahead of server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Europe/Amsterdam' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            const hourOffset = localDateTime.getHours() - converted.getHours()
+            expect([1, 2]).toContain(hourOffset)
+        })
```

Comment:
Just a thought, not a change request: We could test both values separately if we had a way to set the exact date (e. g. by using [`MockDate`](https://github.com/boblauer/MockDate))

Reply:
I've looked into this a bit and for this particular test it would not bring many benefits to the table. We don't really need to be able to mock the current date, but the browser's timezone ID. This is not supported by MockDate.